### PR TITLE
fix: unsupported-features/node-builtins-modules

### PR DESCRIPTION
**Pull request by aladdin-add**
fixes https://github.com/eslint-community/eslint-plugin-n/issues/250

### CODE REVIEW

File: lib/util/check-unsupported-builtins.js [hunk 1]
```diff
@@ -42,15 +43,15 @@ function isSupported({ supported }, configured) {
             ...supported.map(
                 version => `>= ${version} < ${major(version) + 1}`
```
Comment (scagood) [inline lib/util/check-unsupported-builtins.js:44]: Also, thinking about this, could this just be:
```suggestion
                version => `^${version}`
```

File: tests/lib/rules/no-unsupported-features/node-builtins.js [hunk 2]
```diff
@@ -5305,6 +5305,12 @@ new RuleTester({ languageOptions: { sourceType: "module" } }).run(
                         node: { version: "12.11.0" },
                     },
                 },
+
+                // https://github.com/eslint-community/eslint-plugin-n/issues/253
+                {
+                    code: `fetch("/api/xxx");`,
+                    settings: { node: { version: ">=18.2.0" } },
```
Comment (scagood) [inline tests/lib/rules/no-unsupported-features/node-builtins.js:5312]: Does this not need to be `>=18.0.0` to match this?

https://github.com/eslint-community/eslint-plugin-n/blob/a0b45ee9c35279c29d3602609dad3b36d95d6f5f/lib/rules/no-unsupported-features/es-builtins.js#L604-L606

Comment (aladdin-add) [reply tests/lib/rules/no-unsupported-features/node-builtins.js:5312]: well, I think it's worthy adding a few more tests.

Comment (aladdin-add) [reply tests/lib/rules/no-unsupported-features/node-builtins.js:5312]: Found another problem: `fetch` should belong to node-builtins, not es-builtins? 🤔

Comment (scagood) [reply tests/lib/rules/no-unsupported-features/node-builtins.js]: It should probably belong to both, as the es-builtins are all generated 👀 https://github.com/mdn/browser-compat-data + the script here https://github.com/eslint-community/eslint-plugin-n/pull/174#issuecomment-1913255700

Comment (aladdin-add) [reply tests/lib/rules/no-unsupported-features/node-builtins.js]: Technically it is a web api (whatwg). see https://github.com/mdn/browser-compat-data/blob/712e11aa4181a6c92aa552690c93e0354038b705/api/_globals/fetch.json#L3-L6

Comment (aladdin-add) [reply tests/lib/rules/no-unsupported-features/node-builtins.js]: I do think it's another issue (#260), and I just reverted the commit. The pr is now aimed to fix #250.

File: tests/lib/rules/no-unsupported-features/es-builtins.js [hunk 3]
```diff
@@ -2424,6 +2424,23 @@ runTests([
                     },
                 ],
             },
+            // https://github.com/eslint-community/eslint-plugin-n/issues/250
```
[the remark this answers is not in the document]
Comment (scagood) [reply tests/lib/rules/no-unsupported-features/es-builtins.js:2427]: oop, I missed the other comment!

**Pull request merged**